tests/flytekit/unit/core/test_flyte_file.py

```diff
@@ -699,3 +701,26 @@ def test_new_remote_file():
     nf = FlyteFile.new_remote_file(name="foo.txt")
     assert isinstance(nf, FlyteFile)
     assert nf.path.endswith('foo.txt')
+
+
+def test_pickle_flyte_file():
```

Comment:
This test is passing on `master`. I think it is going to this branch:

https://github.com/flyteorg/flytekit/blob/d29e30eedc48f0f3b8d72dd5ea42f9dbdb1f2f13/flytekit/types/file/file.py#L561-L562

and not getting all the way to the `_downloader` yet.

Reply:
Good catch, thanks. I updated the test. We should not only move the downloader to a global scope but also need to ensure it is pickleable somehow. I haven't figured out how to do that yet, so I will revisit it later